src/client/common/terminal/shellDetectors/vscEnvironmentShellDetector.ts

```diff
@@ -23,13 +23,17 @@ export class VSCEnvironmentShellDetector extends BaseShellDetector {
     }
     public identify(
         telemetryProperties: ShellIdentificationTelemetry,
-        _terminal?: Terminal,
+        terminal?: Terminal,
     ): TerminalShellType | undefined {
-        if (!this.appEnv.shell) {
+        const shellPath =
+            terminal?.creationOptions && 'shellPath' in terminal.creationOptions && terminal.creationOptions.shellPath
```

Comment:
```suggestion
            terminal?.creationOptions && terminal.creationOptions.shellPath
```
Shouldn't that just work? if a field is not defined on an object, it is just returns `undefined`.

Reply:
Nope, atleast not for the type check compiler.

`shellPath` property doesn't exist on type of `terminal.creationOptions` (which is `Readonly<TerminalOptions | ExtensionTerminalOptions>`), so we have to narrow down the type first to `Readonly<TerminalOptions>` using this.

Reply:
@karthiknadig Can you also merge this PR as I mentioned here https://github.com/microsoft/vscode-python/pull/16858#issue-702545081